File: tic tac toe.py
import sys

a = [
    ['N','N','N'],
    ['N','N','N'],
    ['N','N','N']
]

First3Vertically = []
First3Horizontally = []

GameOver = False
Row = None
Colum = None

Turn = "O"
Winner = None




def FindFirst3Vertically(row):
    for i in range(3):
        First3Vertically.append(a[row][i])
        if len(First3Vertically) == 3:

            if First3Vertically[0] == First3Vertically[1] == First3Vertically[2]:
                if "N" not in First3Vertically:
                    print(First3Vertically[0][0],"Is Winner")
                    sys.exit()
                    return True


            First3Vertically.clear()

def FindFirst3Horizontally(item):
   for i in range(3):
        First3Horizontally.append(a[i][item])

        if len(First3Horizontally) == 3:

            if First3Horizontally[0] == First3Horizontally[1] == First3Horizontally[2]:
                if "N" not in First3Horizontally:
                    print(First3Horizontally[0][0],"Is winner")
                    sys.exit()
                    return True


            First3Horizontally.clear()


def CheckDiagnoally():

    if a[2][0] != "N" and a[1][1] != "N" and a[0][2] != "N":
        if a[2][0] == a[1][1] == a[0][2]:
            print(a[1][1],"is the winner")
            sys.exit()

    if a[0][0] != "N" and a[1][1] != "N" and a[2][2] != "N":
        if a[0][0] == a[1][1] == a[2][2]:
            print(a[1][1],"is the winner")
            sys.exit()




def CheckWinner():
    # print("running")
    for i in range(3):
        VerticalTestResult = FindFirst3Vertically(i)

    for i in range(3):
        HorizontalTestResult = FindFirst3Horizontally(i)
        # print(HorizontalTestResult)

    c = CheckDiagnoally()

    if not HorizontalTestResult  or not VerticalTestResult or not c :
        if "N" not in a[0] and a[1] and a[2]:
            print("Draw")
            sys.exit()




def DrawBoard():
    for i in range(len(a)):
        print(a[i])



DrawBoard()
print("Quick Note : Indexing Starts From 0\n")
while not GameOver:

    CheckWinner()

    if Turn == "O":
        print(Turn,"'s","Turn\n")
        Row = int(input("Row: "))
        Colum = int(input("Colum: "))
        if a[Row][Colum] == "N":
            a[Row][Colum] = "O"
            Turn = "X"

        else:
            print("That Square Is Occupied")
            Turn = "O"

        DrawBoard()


    CheckWinner()

    if Turn == "X":
        print(Turn,"'s","Turn\n")
        Row = int(input("Row: "))
        Colum = int(input("Colum: "))
        Turn = "O"
        if a[Row][Colum] == "N":
            a[Row][Colum] = "X"
            Turn = "O"

        else:
            print("That Square is Occupied")
            Turn = "X"

        DrawBoard()

    CheckWinner()

File: TicTacToe.c
#include <stdio.h>
#include<stdlib.h>


char board[3][3] = {{'.', '.', '.'}
                    ,{'.', '.', '.'}, 
                    {'.', '.', '.'}};
char b[3];
int PrintBoard(void);
int fetch(int row, char b[]);
int fetch2(int row, char b[]);
int draw(void);
int CheckWinner();

int main(void)
{
    PrintBoard();

    int row, colum;
    char turn;

    turn = 'O';
    
    while (1)
    {      
        
        
        CheckWinner();
        


        if (turn == 'O')
        {
            printf("%c%s\n", turn, "'s turn");
            printf("\n");
            printf("Row: ");
            scanf("%d", &row);

            printf("Colum: ");
            scanf("%d", &colum);
            printf("\n");

            if (board[row][colum] == '.')
            {
                board[row][colum] = 'O';
                PrintBoard();
                turn = 'X';
            }
            else
            {
                printf("That cell is occupied\n");
                PrintBoard();
                turn = 'O';
            }
        }

        CheckWinner();

        if (turn == 'X')
        {
            printf("%c%s\n", turn, "'s turn");

            printf("Row: ");
            scanf("%d", &row);

            printf("Colum: ");
            scanf("%d", &colum);
            printf("\n");
            if (board[row][colum] == '.')
            {
                board[row][colum] = 'X';
                PrintBoard();
                turn = 'O';
            }
            else
            {
                printf("That cell Is occupied\n");
                PrintBoard();
                turn = 'X';
            }
        }
        
        
        CheckWinner();
        
    }
    return 0;
}

int PrintBoard(void)
{
    for (int i = 0; i < 3; i++)
    {
        for (int j = 0; j < 3; j++)
        {
            printf("%c\t", board[i][j]);
        }
        printf("\n\n");
    }

    
    return 0;
}

int fetch(int row, char b[])
{
    for (int i = 0; i < 3; i++)
    {
        b[i] = board[row][i];
    }

    if (b[0] != '.' || b[1] != '.' || b[2] != '.')
    {
        if (b[0] == b[1])
        {
            if (b[1] == b[2])
            {
                printf("%c%s\n", b[0], " is the winner");
                exit(0);
                return 1;
            }
        }
    }

}
int fetch2(int row, char b[])
{
    for (int i = 0; i < 3; i++)
    {
        b[i] = board[i][row];
    }

    if (b[0] != '.' || b[1] != '.' || b[2] != '.')
    {
        if (b[0] == b[1])
        {
            if (b[1] == b[2])
            {
                printf("%c%s\n", b[0], " is the winner");
                exit(0);
                
                return 1;
            }

        }
    }

}

int draw(void)
{   
    int counter = 0;
    for (int i = 0; i<3; i++)
    {
        for (int j = 0; j<3; j++)
        {
            if (board[i][j] !='.')
            {
                counter++;
            }
        }
    }

    return counter;
}

int CheckWinner()
{      
    int check;
    for (int i = 0; i < 3; i++)
    {   
        check = fetch2(i, b);
        
    }

    for (int i = 0; i < 3; i++)
    {
        check = fetch(i, b);
    }

    if (check !=1 && draw() == 9)
    {
        printf("Draw\n");
        exit(0);
    }


    

   
}
